lib/puppet/provider/templates/webadministration/_getvirtualdirectories.ps1.erb

```diff
@@ -1,24 +1,25 @@
 Get-WebVirtualDirectory | % {
-  $path         = $_.Path
-  $physicalpath =  [string]$_.PhysicalPath
+  $application  = '/'
+  $name         = [string]$_.Path
+  $physicalpath = [string]$_.PhysicalPath
+  $sitename     = ''
 
-  if($path -match "\/.*\/.*"){
-    $application = $_.Path.Split("/")[1]
-    $name        = $_.Path.Split("/")[2]
-  }else{
-    $name        = $_.Path.Trim("/")
-    $application = "/"
+  if($_.ItemXPath -Match "application\[\@path\=(.*?)\]"){
+    $application = $matches[1].Replace("'",'')
   }
 
-  if($_.ItemXPath -match ".*name\=(.*) and @id=(.*)"){
-    $site = $matches[1].Replace("'","")
+  $name = $name -Replace "^/", ''
+  $name = $name -Replace "/", '\'
+
+  if($_.ItemXPath -Match "site\[\@name\=(.*?) and"){
+    $sitename = $matches[1].Replace("'",'')
   }
 
   New-Object -TypeName PSObject -Property @{
+    application  = $application
```

Comment:
I'd like to make the case that including initializing all for variables at the top and processing them in a specific order.